### PR TITLE
Fix linear

### PR DESCRIPTION
**Pull request by dan-garvey**
Fixes punet regression from linear.py
I went a little crazy with the fake-quant arg and the logic was
ugly. Now the fake_quant arg only affects q_input. The purpose in having
it at all is to allow exporting and fake eager using the same irpa file

### CODE REVIEW

**Comment by nithinsubbiah** [comment]
Looks like sharktank integration test is still failing

File: .github/workflows/ci-sharktank.yml [hunk 1]
```diff
@@ -22,6 +22,31 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  test_punet:
```
Comment (nithinsubbiah) [inline .github/workflows/ci-sharktank.yml:25]: We should also change model version to this (https://github.com/nod-ai/shark-ai/pull/569/files) to reflect the production model

Comment (dan-garvey) [reply .github/workflows/ci-sharktank.yml:25]: ah my bad good catch

**Review by nithinsubbiah — APPROVED** [review]
Thanks for fixing the export. I don't know if the llama data dependent test failure is expected, contingent on that good to merge.

**Comment by dan-garvey** [comment]
I'm not sure this is okay to merge as is, it takes over an hour to run the punet

**Pull request merged**